OMP compilation error in registration test

## Your Environment
* Operating System and version: Linux
* Compiler: GCC 9.1, OpenMP 8.0.1
* PCL Version: master

## Expected Behavior
No compile errors

## Current Behavior
```
Building CXX object test/registration/CMakeFiles/test_fpcs_ia.dir/test_fpcs_ia.cpp.o
FAILED: test/registration/CMakeFiles/test_fpcs_ia.dir/test_fpcs_ia.cpp.o
/usr/bin/c++  -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB
-DvtkRenderingContext2D_AUTOINIT="1(vtkRenderingContextOpenGL2)"
-DvtkRenderingCore_AUTOINIT="3(vtkInteractionStyle,vtkRenderingFreeType,
vtkRenderingOpenGL2)" -I/usr/include/freetype2 -I/usr/include/double-conversion
-Iinclude -I../global_tests/include -I../registration/include -I../common/include
-I../octree/include -I../kdtree/include -I../search/include -I../sample_consensus/include
-I../features/include -I../filters/include -I../io/include -isystem /usr/include/eigen3
-isystem ../recognition/include/pcl/recognition/3rdparty -isystem /usr/include/vtk
-isystem /store/workspace/googletest/googletest/include -isystem
/store/workspace/googletest/googletest -isystem /usr/include/qt -isystem
/usr/include/qt/QtWidgets -isystem /usr/include/qt/QtGui -isystem
/usr/include/qt/QtCore -isystem /usr/lib/qt/mkspecs/linux-g++ -Wabi=11 -Wall
-Wextra -Wno-unknown-pragmas -fno-strict-aliasing -Wno-format-extra-args
-Wno-sign-compare -Wno-invalid-offsetof -Wno-conversion -march=native
-msse4.2 -mfpmath=sse -fopenmp  -O2 -g -DNDEBUG   -DBOOST_DISABLE_ASSERTS
-DEIGEN_NO_DEBUG -fPIC -std=c++14 -MD -MT test/registration/CMakeFiles/
test_fpcs_ia.dir/test_fpcs_ia.cpp.o -MF test/registration/CMakeFiles/test_fpcs_ia.dir
/test_fpcs_ia.cpp.o.d -o test/registration/CMakeFiles/test_fpcs_ia.dir/test_fpcs_ia.cpp.o
 -c ../test/registration/test_fpcs_ia.cpp
In file included from ../registration/include/pcl/registration/ia_fpcs.h:567,
                 from ../test/registration/test_fpcs_ia.cpp:42:
../registration/include/pcl/registration/impl/ia_fpcs.hpp: In function ‘float
pcl::getMeanPointDensity(const typename pcl::PointCloud<PointT>::ConstPtr&,
const std::vector<int>&, float, int) [with PointT = pcl::PointXYZ]’:
../registration/include/pcl/registration/impl/ia_fpcs.hpp:108:58: error: ‘s’ not
specified in enclosing ‘parallel’
  108 |     tree.nearestKSearch (cloud->points[indices[rand () % s]], 2, ids, dists_sqr);
      |                                                          ^
../registration/include/pcl/registration/impl/ia_fpcs.hpp:100:9: error: enclosing ‘parallel’
  100 | #pragma omp parallel for \
      |         ^~~
../registration/include/pcl/registration/impl/ia_fpcs.hpp:109:24: error: ‘max_dist_sqr’
not specified in enclosing ‘parallel’
  109 |     if (dists_sqr[1] < max_dist_sqr)
      |                        ^~~~~~~~~~~~
../registration/include/pcl/registration/impl/ia_fpcs.hpp:100:9: error: enclosing ‘parallel’
  100 | #pragma omp parallel for \
      |         ^~~
```

--- a/registration/include/pcl/registration/impl/ia_fpcs.hpp
+++ b/registration/include/pcl/registration/impl/ia_fpcs.hpp
@@ -62,7 +62,7 @@ pcl::getMeanPointDensity (const typename pcl::PointCloud<PointT>::ConstPtr &clou
 #ifdef _OPENMP
 #pragma omp parallel for \
   reduction (+:mean_dist, num) \
-  private (ids, dists_sqr) shared (tree, cloud) \
+  private (ids, dists_sqr) shared (tree, cloud) firstprivate (s, max_dist_sqr) \
   default (none)num_threads (nr_threads)
 #endif
 
@@ -99,7 +99,7 @@ pcl::getMeanPointDensity (const typename pcl::PointCloud<PointT>::ConstPtr &clou
 #ifdef _OPENMP
 #pragma omp parallel for \
   reduction (+:mean_dist, num) \
-  private (ids, dists_sqr) shared (tree, cloud, indices)    \
+  private (ids, dists_sqr) shared (tree, cloud, indices) firstprivate (s, max_dist_sqr) \
   default (none)num_threads (nr_threads)
 #endif
 
@@ -763,7 +763,7 @@ pcl::registration::FPCSInitialAlignment <PointSource, PointTarget, NormalT, Scal
   pcl::Correspondences &correspondences)
 {
   // calculate centroid of base and target
-  Eigen::Vector4f centre_base, centre_match;
+  Eigen::Vector4f centre_base {0, 0, 0, 0}, centre_match {0, 0, 0, 0};
   pcl::compute3DCentroid (*target_, base_indices, centre_base);
   pcl::compute3DCentroid (*input_, match_indices, centre_match);
 

